core/src/main/java/org/elasticsearch/threadpool/ThreadPool.java

```diff
@@ -169,7 +169,7 @@ public ThreadPool(final Settings settings, final ExecutorBuilder<?>... customBui
         final int genericThreadPoolMax = boundedBy(4 * availableProcessors, 128, 512);
         builders.put(Names.GENERIC, new ScalingExecutorBuilder(Names.GENERIC, 4, genericThreadPoolMax, TimeValue.timeValueSeconds(30)));
         builders.put(Names.INDEX, new FixedExecutorBuilder(settings, Names.INDEX, availableProcessors, 200));
```

Comment:
`INDEX` TP is currently only used by `TransportUpdateAction`. I agree that we should remove it or at least rename it to better reflect where it is used.